test/github/test_logic.py

```diff
@@ -746,6 +746,34 @@ def test_rerun_stale_checks_on_automerge_pr(
         )
         mock_merge_pull_request.assert_not_called()
 
+    @patch("src.github.logic.SGTM_FEATURE__CHECK_RERUN_ON_APPROVAL_ENABLED", True)
+    @patch("src.github.logic.SGTM_FEATURE__CHECK_RERUN_THRESHOLD_HOURS", 1)
+    @patch(
+        "src.github.logic.SGTM_FEATURE__CHECK_RERUN_BASE_REF_NAMES",
+        ["main"],
+    )
+    def test_no_rerun_when_check_run_not_complete(self, mock_rerequest_check_run):
+        check_run = build(builder.check_run().completed_at(None))
```

Comment:
hmm surprisingly, there's no type error from mypy. The Check Run Builder's method for `completed_at` is `Union[str, datetime]` so you'll want to update that to be optional as well. Just for correctness

Reply:
updated the types!